File: src/main/java/uk/co/eelpieconsulting/osm/nominatim/json/JsonDeserializer.java
package uk.co.eelpieconsulting.osm.nominatim.json;

import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.stereotype.Component;
import uk.co.eelpieconsulting.osm.nominatim.model.Place;

import java.io.IOException;

@Component
public class JsonDeserializer {

  private final ObjectMapper mapper;

  {
    this.mapper = new ObjectMapper();
    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
  }

  public Place deserializePlace(String sourceAsString) throws IOException {
    return mapper.readValue(sourceAsString, Place.class);
  }

}

File: src/main/kotlin/uk/co/eelpieconsulting/osm/nominatim/json/JsonDeserializer.kt
package uk.co.eelpieconsulting.osm.nominatim.json

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import com.fasterxml.jackson.module.kotlin.registerKotlinModule
import org.springframework.stereotype.Component
import uk.co.eelpieconsulting.osm.nominatim.model.Place

@Component
class JsonDeserializer {

    private val mapper: ObjectMapper

    init {
        this.mapper = ObjectMapper().registerKotlinModule()
    }

    fun deserializePlace(sourceAsString: String): Place {
        return mapper.readValue<Place>(sourceAsString)
    }

}
